app/javascript/chat/chat.jsx

```diff
@@ -1314,21 +1302,24 @@ export class Chat extends Component {
           {invitesButton}
           {joiningRequestButton}
           <div className="chat__channeltypefilter">
-            {this.renderChannelFilterButton(
-              'all',
-              'all',
-              state.channelTypeFilter,
-            )}
-            {this.renderChannelFilterButton(
-              'direct',
-              'direct',
-              state.channelTypeFilter,
-            )}
-            {this.renderChannelFilterButton(
-              'invite_only',
-              'group',
-              state.channelTypeFilter,
-            )}
+            <ChannelFilterButton
+              type={'all'}
+              name={'all'}
+              active={state.channelTypeFilter}
```

Comment:
(refactor / blocking) We're currently passing a string of the currently active type as the `active` prop, but I think it should be refactored to be a boolean - it would simplify the new `ChannelFilterButton` component

```suggestion
              active={state.channelTypeFilter === 'all'}
```